diff --color -rupN old/lib/config.sub new/lib/config.sub
--- old/lib/config.sub	2021-07-07 13:39:57.000000000 +0800
+++ new/lib/config.sub	2021-08-04 12:14:55.797433028 +0800
@@ -1716,7 +1716,7 @@ case $os in
 	     | os9* | macos* | osx* | ios* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
-	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
+	     | aos* | aros* | cloudabi* | sortix* | twizzler* | lyos* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
