diff -rupN old/build-aux/config.sub new/build-aux/config.sub
--- old/build-aux/config.sub	2018-06-24 12:52:06.000000000 +0800
+++ new/build-aux/config.sub	2018-08-27 18:00:49.229305745 +0800
@@ -1376,7 +1376,7 @@ case $os in
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd*)
+	     | midnightbsd* | lyos* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
