diff -rupN old/config.sub new/config.sub
--- old/config.sub	2019-01-03 19:28:27.000000000 +0800
+++ new/config.sub	2021-04-20 11:10:25.329704989 +0800
@@ -2,7 +2,7 @@
 # Configuration validation subroutine script.
 #   Copyright 1992-2019 Free Software Foundation, Inc.
 
-timestamp='2019-01-01'
+timestamp='2020-06-28'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
@@ -1361,6 +1361,7 @@ case $os in
 	     | os2* | vos* | palmos* | uclinux* | nucleus* \
 	     | morphos* | superux* | rtmk* | windiss* \
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
+         | lyos* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten*)
diff -rupN old/gcc/config/lyos.h new/gcc/config/lyos.h
--- old/gcc/config/lyos.h	1970-01-01 08:00:00.000000000 +0800
+++ new/gcc/config/lyos.h	2021-04-20 11:10:25.329704989 +0800
@@ -0,0 +1,27 @@
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS() \
+  do { \
+    builtin_define_std ("lyos"); \
+    builtin_define_std ("unix"); \
+    builtin_assert ("system=lyos"); \
+    builtin_assert ("system=unix"); \
+  } while (0);
+
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "%{!shared:crt0.o%s} \
+   crti.o%s %{static:crtbeginT.o%s;shared|pie:crtbeginS.o%s;:crtbegin.o%s}"
+
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "crtend.o%s crtn.o%s"
+
+#undef  LINK_SPEC
+#define LINK_SPEC "\
+  %{shared:-shared -dynamic-linker " "/lib/ld-lyos.so" "} \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{!dynamic-linker:-dynamic-linker " "/lib/ld-lyos.so" "}} \
+    %{static:-static}}"
+
+#undef  LIB_SPEC
+#define LIB_SPEC "%{pthread:-lpthread} %{shared:-lc} %{!shared:%{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
diff -rupN old/gcc/config.gcc new/gcc/config.gcc
--- old/gcc/config.gcc	2019-08-04 04:16:22.000000000 +0800
+++ new/gcc/config.gcc	2021-04-20 11:10:25.329704989 +0800
@@ -970,6 +970,12 @@ case ${target} in
       ;;
   esac
   ;;
+*-*-lyos*)
+  extra_options="$extra_options gnu-user.opt"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  ;;  
 *-*-elf|arc*-*-elf*)
   # Assume that newlib is being used and so __cxa_atexit is provided.
   default_use_cxa_atexit=yes
@@ -1135,6 +1141,11 @@ arm-wrs-vxworks|arm-wrs-vxworks7)
         tm_file="elfos.h arm/elf.h arm/aout.h ${tm_file} vx-common.h vxworks.h arm/vxworks.h"
 	target_cpu_cname="arm8"
 	;;
+arm-*-lyos*)
+	tm_file="dbxelf.h elfos.h newlib-stdint.h arm/unknown-elf.h arm/elf.h arm/aout.h arm/arm.h lyos.h"
+	tmake_file="arm/t-arm arm/t-arm-elf"
+    # thread_file="posix"
+	;;    
 arm*-*-freebsd*)                # ARM FreeBSD EABI
 	tm_file="dbxelf.h elfos.h ${fbsd_tm_file} arm/elf.h"
 	case $target in
@@ -1612,6 +1623,12 @@ hppa[12]*-*-hpux11*)
 i[34567]86-*-darwin1[89]*)
 	echo "Error: 32bit target is not supported after Darwin17" 1>&2
 	;;
+i[34567]86-*-lyos*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h glibc-stdint.h elfos.h i386/i386elf.h lyos.h"
+	tmake_file="${tmake_file} i386/t-i386elf t-svr4 i386/t-crtstuff"
+	use_fixproto=yes
+    # thread_file="posix"
+	;;    
 i[34567]86-*-darwin*)
 	need_64bit_isa=yes
 	# Baseline choice for a machine that allows m64 support.
@@ -2198,6 +2215,14 @@ microblaze*-*-elf)
 	c_target_objs="${c_target_objs} microblaze-c.o"
 	cxx_target_objs="${cxx_target_objs} microblaze-c.o"
 	tmake_file="${tmake_file} microblaze/t-microblaze"
+	;;
+riscv*-*-lyos*)
+	tm_file="elfos.h newlib-stdint.h ${tm_file} riscv/elf.h lyos.h"
+	tmake_file="${tmake_file} riscv/t-riscv"
+	# Force .init_array support.  The configure script cannot always
+	# automatically detect that GAS supports it, yet we require it.
+	gcc_cv_initfini_array=yes
+    # thread_file="posix"
         ;;
 riscv*-*-linux*)
 	tm_file="elfos.h gnu-user.h linux.h glibc-stdint.h ${tm_file} riscv/linux.h"
diff -rupN old/gcc/diagnostic.c new/gcc/diagnostic.c
--- old/gcc/diagnostic.c	2019-01-01 20:31:55.000000000 +0800
+++ new/gcc/diagnostic.c	2021-04-20 11:14:04.623046691 +0800
@@ -35,9 +35,7 @@ along with GCC; see the file COPYING3.
 #include "selftest.h"
 #include "selftest-diagnostic.h"
 
-#ifdef HAVE_TERMIOS_H
-# include <termios.h>
-#endif
+#include <termios.h>
 
 #ifdef GWINSZ_IN_SYS_IOCTL
 # include <sys/ioctl.h>
diff -rupN old/libgcc/config.host new/libgcc/config.host
--- old/libgcc/config.host	2019-01-17 20:29:13.000000000 +0800
+++ new/libgcc/config.host	2021-04-20 11:10:25.329704989 +0800
@@ -101,6 +101,9 @@ arc*-*-*)
 arm*-*-*)
 	cpu_type=arm
 	;;
+arm-*-lyos*)
+	extra_parts="crtbegin.o crtend.o"
+	;;    
 avr-*-*)
 	cpu_type=avr
 	;;    
@@ -812,6 +815,10 @@ x86_64-*-mingw*)
 		extra_parts="$extra_parts vtv_start.o vtv_end.o vtv_start_preinit.o vtv_end_preinit.o"
 	fi
 	;;
+i[34567]86-*-lyos*)
+	tmake_file="$tmake_file i386/t-i386elf t-svr4 i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	extra_parts="$extra_parts crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o"
+	;;    
 ia64*-*-elf*)
 	extra_parts="$extra_parts crtbeginS.o crtendS.o crtfastmath.o"
 	tmake_file="ia64/t-ia64 ia64/t-ia64-elf ia64/t-eh-ia64 t-crtfm t-softfp-tf ia64/t-softfp t-softfp ia64/t-softfp-compat"
diff -rupN old/libstdc++-v3/crossconfig.m4 new/libstdc++-v3/crossconfig.m4
--- old/libstdc++-v3/crossconfig.m4	2019-03-12 00:28:11.000000000 +0800
+++ new/libstdc++-v3/crossconfig.m4	2021-04-20 11:10:25.333038323 +0800
@@ -200,6 +200,12 @@ case "${host}" in
     AC_CHECK_FUNCS(sockatmark)
     AM_ICONV
     ;;
+  *-lyos*)
+    AC_CHECK_HEADERS([sys/types.h local.h float.h])
+    GLIBCXX_CHECK_BUILTIN_MATH_SUPPORT
+    GLIBCXX_CHECK_COMPLEX_MATH_SUPPORT
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;    
   *-mingw32*)
     GLIBCXX_CHECK_LINKER_FEATURES
     GLIBCXX_CHECK_MATH_SUPPORT
