diff -rupN origin/config.sub new/config.sub
--- origin/config.sub	2012-12-08 08:02:34.000000000 +0800
+++ new/config.sub	2015-02-18 16:11:14.172984536 +0800
@@ -1379,6 +1379,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
+	      | -lyos* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
diff -rupN origin/newlib/configure.host new/newlib/configure.host
--- origin/newlib/configure.host	2012-12-05 05:43:20.000000000 +0800
+++ new/newlib/configure.host	2015-02-19 12:37:55.720620131 +0800
@@ -372,6 +372,9 @@ case "${host}" in
   arc-*-*)
 	sys_dir=arc
 	;;
+  arm-*-lyos*)
+	sys_dir=lyos
+	;;
   arm*-*-*)
 	sys_dir=arm
 	if [ "x${newlib_may_supply_syscalls}" = "xno" ] ; then
@@ -419,6 +422,10 @@ case "${host}" in
   h8500-*-elf*)
 	sys_dir=h8500hms
 	;;
+  i[34567]86-*-lyos*)
+	sys_dir=lyos
+	newlib_cflags="${newlib_cflags} -D_NO_GETPWENT"
+	;;
   i[34567]86-*-rdos*)
 	sys_dir=rdos
 	newlib_cflags="${newlib_cflags} -DMISSING_SYSCALL_NAMES"
@@ -526,6 +533,9 @@ esac
 # THIS TABLE IS ALPHA SORTED.  KEEP IT THAT WAY.
 
 case "${host}" in
+  *-*-lyos)
+	newlib_cflags="${newlib_cflags} -DMISSING_SYSCALL_NAMES -DSIGNAL_PROVIDED"
+	;;
   *-*-cygwin*)
 	test -z "$cygwin_srcdir" && cygwin_srcdir=`cd ${srcdir}/../winsup/cygwin; pwd`
 	export cygwin_srcdir
diff -rupN origin/newlib/libc/include/err.h new/newlib/libc/include/err.h
--- origin/newlib/libc/include/err.h	1970-01-01 08:00:00.000000000 +0800
+++ new/newlib/libc/include/err.h	2015-04-12 14:16:47.271253864 +0800
@@ -0,0 +1,16 @@
+#ifndef _ERR_H_
+#define _ERR_H_
+
+#include <stdarg.h>
+
+void err(int eval, const char *fmt, ...);
+void errx(int eval, const char *fmt, ...);
+void warn(const char *fmt, ...);
+void warnx(const char *fmt, ...);
+
+void verr(int eval, const char *fmt, va_list args);
+void verrx(int eval, const char *fmt, va_list args);
+void vwarn(const char *fmt, va_list args);
+void vwarnx(const char *fmt, va_list args);
+
+#endif
diff -rupN origin/newlib/libc/include/mntent.h new/newlib/libc/include/mntent.h
--- origin/newlib/libc/include/mntent.h	1970-01-01 08:00:00.000000000 +0800
+++ new/newlib/libc/include/mntent.h	2015-02-24 20:30:07.062823618 +0800
@@ -0,0 +1,22 @@
+#ifndef _MNTENT_H_
+#define _MNTENT_H_
+
+#define _PATH_MNTTAB "/etc/fstab"
+#define _PATH_MOUNTED "/etc/mtab"
+
+struct mntent {
+	char    *mnt_fsname;    /* name of mounted file system */
+	char    *mnt_dir;       /* file system path prefix */
+	char    *mnt_type;      /* mount type (see mntent.h) */
+	char    *mnt_opts;      /* mount options (see mntent.h) */
+	int     mnt_freq;       /* dump frequency in days */
+	int     mnt_passno;     /* pass number on parallel fsck */
+};
+
+FILE *setmntent(const char *filename, const char *type);
+struct mntent *getmntent(FILE *stream);
+int addmntent(FILE *stream, const struct mntent *mnt);
+int endmntent(FILE *streamp);
+char *hasmntopt(const struct mntent *mnt, const char *opt);
+
+#endif
diff -rupN origin/newlib/libc/machine/i386/setjmp.S new/newlib/libc/machine/i386/setjmp.S
--- origin/newlib/libc/machine/i386/setjmp.S	2012-03-10 04:26:36.000000000 +0800
+++ new/newlib/libc/machine/i386/setjmp.S	2015-02-18 16:12:20.577313814 +0800
@@ -75,7 +75,6 @@ SYM (longjmp):
 
 	movl	24(edi),ebp
 
-       __CLI
 	movl	28(edi),esp
 	
 	pushl	32(edi)	
@@ -86,6 +85,5 @@ SYM (longjmp):
 	movl	12(edi),edx
 	movl	16(edi),esi
 	movl	20(edi),edi
-       __STI
 
 	ret
diff -rupN origin/newlib/libc/sys/configure.in new/newlib/libc/sys/configure.in
--- origin/newlib/libc/sys/configure.in	2012-02-22 06:34:31.000000000 +0800
+++ new/newlib/libc/sys/configure.in	2015-02-18 16:11:14.176984554 +0800
@@ -46,6 +46,7 @@ if test -n "${sys_dir}"; then
 	sysvnecv70) AC_CONFIG_SUBDIRS(sysvnecv70) ;;
 	tic80) AC_CONFIG_SUBDIRS(tic80) ;;
 	w65) AC_CONFIG_SUBDIRS(w65) ;;
+	lyos) AC_CONFIG_SUBDIRS(lyos) ;;
 	z8ksim) AC_CONFIG_SUBDIRS(z8ksim) ;;
   esac;
 fi
